src/python/module/smartredis/configoptions.py

```diff
@@ -25,20 +25,48 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import typing as t
+from functools import wraps
+from uuid import uuid4
 
 from .error import RedisRuntimeError
 from .smartredisPy import PyConfigOptions
 from .util import exception_handler, typecheck
 
-_NOT_FACTORY = (
-    "Method called on a ConfigOptions object not created from a factory method"
-)
+
+class _Managed:
+    """Marker class identifying factory-created objects"""
+
+
+def create_config_options(base: t.Type[t.Any]) -> t.Any:
+    """Factory method for creating managed instances"""
+    bases = (_Managed, base)
+    unique_key = str(uuid4()).split("-", 1)[0]
+    class_name = "".join(base.__name__ for base in bases) + unique_key
```

Comment:
I'd still really like some extra documentation on these two lines for if I have to come back and do maintenance on it in 6 months

Reply:
please see added comments & nested method